src/js/lib/safe.js

```diff
@@ -44,15 +36,21 @@ var makeTranslateStack = function(stackLineRegExp, translateLine) {
         line = lines[i] = translateLine.apply(this, m);
       }
       if (line.match(module.filename)) {
-        lines.splice(--i, 2);
+        lines.splice(i, 1);
```

Comment:
hard to review those changes without knowing what the problems were. In general I would say this is the kind of functions that would deserve unit test. Or at least clear comments on expected in/out.